### DISCUSSION TITLE
How can I find loop start point if loop's end point is exactly where track ends?

### DISCUSSION THREAD

**Discussion by Shararamosh** [arkrow/PyMusicLooper#33] [General]
I have a bunch of game tracks that are merged intro and loop parts - the loop start point is undefined, but loop end point is exactly at the end of track. I can approximately find where the loop should start, however putting these 2 points with --approx-loop-position (and even with --brute-force) hasn't given me a good result. Is it possible to find best loop start point in this case?

**Comment by arkrow** [comment]
I generated some samples to reproduce by using `split-audio` and re-running on the merged intro+loop sections, though I found that PyMusicLooper can still extract close to the same loop points from the initial split. Can you try running the `--brute-force` option with `--disable-pruning`? I had better results with that flag combination. If there are too many loop points, they are usually pruned by default based on the calculated note/loudness difference to reduce processing time and improve end results, but in some cases, pruning gets rid of perceptually good loop points.

Let me know what your experience is if you try it. The `--brute-force --disable-pruning` flag combination should output all potential loop pairs that pass the initial note/loudness check without filtering them further. If it doesn't work, I can consider opening up additional advanced configuration options for the loop search algorithm, namely: max allowed note difference, max allowed loudness difference, and the loop scoring test duration.
